cli/azd/internal/repository/app_init.go

```diff
@@ -472,23 +485,74 @@ func (i *Initializer) prjConfigFromDetect(
 					config.Resources["postgres"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbPostgres,
 						Name: spec.DbPostgres.DatabaseName,
+						Props: project.PostgresProps{
+							DatabaseName: spec.DbPostgres.DatabaseName,
+							AuthType: decideAuthType(
+								spec.DbPostgres.AuthUsingManagedIdentity,
+								spec.DbPostgres.AuthUsingUsernamePassword,
+								false),
+						},
 					}
 				case appdetect.DbMySql:
 					config.Resources["mysql"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbMySQL,
 						Props: project.MySQLProps{
 							DatabaseName: spec.DbMySql.DatabaseName,
-							AuthType:     "managedIdentity",
+							AuthType: decideAuthType(
+								spec.DbMySql.AuthUsingManagedIdentity,
+								spec.DbMySql.AuthUsingUsernamePassword,
+								false),
 						},
 					}
 				case appdetect.DbRedis:
 					config.Resources["redis"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbRedis,
 					}
+				case appdetect.DbCosmos:
+					cosmosDBProps := project.CosmosDBProps{
+						DatabaseName: spec.DbCosmos.DatabaseName,
+					}
+					for _, container := range spec.DbCosmos.Containers {
+						cosmosDBProps.Containers = append(cosmosDBProps.Containers, project.CosmosDBContainerProps{
+							ContainerName:     container.ContainerName,
+							PartitionKeyPaths: container.PartitionKeyPaths,
+						})
+					}
+					config.Resources["cosmos"] = &project.ResourceConfig{
+						Type:  project.ResourceTypeDbCosmos,
+						Props: cosmosDBProps,
+					}
+				}
+
+			}
+			for _, azureDep := range prj.AzureDeps {
+				switch azureDep.(type) {
+				case appdetect.AzureDepServiceBus:
+					config.Resources["servicebus"] = &project.ResourceConfig{
+						Type: project.ResourceTypeMessagingServiceBus,
+						Props: project.ServiceBusProps{
+							Queues: spec.AzureServiceBus.Queues,
+							IsJms:  spec.AzureServiceBus.IsJms,
+							AuthType: decideAuthType(
+								spec.AzureServiceBus.AuthUsingManagedIdentity,
+								false,
+								spec.AzureServiceBus.AuthUsingConnectionString),
+						},
+					}
+				case appdetect.AzureDepEventHubs:
+					config.Resources["eventhubs"] = &project.ResourceConfig{
+						Type: project.ResourceTypeMessagingEventHubs,
+						Props: project.EventHubsProps{
+							EventHubNames: spec.AzureEventHubs.EventHubNames,
+							AuthType: decideAuthType(
+								spec.AzureServiceBus.AuthUsingManagedIdentity,
+								false,
+								spec.AzureServiceBus.AuthUsingConnectionString),
+						},
+					}
```

Comment:
Converting from `spec.AzureEventHubs` to `project.EventHubsProps` happened more than once, how about create a util method for it?

Reply:
Same for this one, I prefer to do this kind of refactor later.